program-runtime/src/loaded_programs.rs

```diff
@@ -94,13 +101,32 @@ impl LoadedProgram {
         deployment_slot: Slot,
         elf_bytes: &[u8],
         account_size: usize,
+        metrics: &mut LoadProgramMetrics,
     ) -> Result<Self, EbpfError> {
         let program = if bpf_loader_deprecated::check_id(loader_key) {
```

Comment:
Lots of redundant code, only needs to be different at: `let program_type = LoadedProgramType::`, the rest could be shared.

Reply:
Done